onnxruntime/core/providers/cuda/tensor/identity_op.h

```diff
@@ -15,34 +16,57 @@ class IdentityOp final : public CudaKernel {
   }
 
   Status ComputeInternal(OpKernelContext* context) const override {
-    const Tensor* X = context->Input<Tensor>(0);
-    if (X == nullptr) return Status(common::ONNXRUNTIME, common::FAIL, "input count mismatch");
-    const TensorShape& shape = X->Shape();
-    Tensor* Y = context->Output(0, shape);
-    auto X_type = X->DataType();
+    auto X_ml_type = context->InputType(0);
+    if (DataTypeImpl::GetType<Tensor>() == X_ml_type) {
```

Comment:
>DataTypeImpl::GetType<Tensor>() == X_ml_type [](http://example.com/codeflow?start=8&length=44)

This only covers a base class. use:
if(X_ml_type->IsTensor())

Reply:
Thanks - add corresponding interface in provider interface.

Reply:
Good point, thanks!